enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/StoreCopyRequestHandler.java

```diff
@@ -84,9 +85,10 @@ protected void channelRead0( ChannelHandlerContext ctx, T request ) throws Excep
             {
                 responseStatus = StoreCopyFinishedResponse.Status.E_STORE_ID_MISMATCH;
             }
-            else if ( !isTransactionWithinReach( request.requiredTransactionId(), checkpointerSupplier.get() ) )
+            else if ( !isTransactionWithinReach( request.requiredTransactionId() ) )
             {
                 responseStatus = StoreCopyFinishedResponse.Status.E_TOO_FAR_BEHIND;
+                asyncCheckpoint();
```

Comment:
I think we need to throttle this somehow. Maybe have a single thread pool that is shared for all channels to avoid spawning multiple threads that are only waiting for a check point to complete.

Reply:
This has been addressed with a single thread pool coming in from CoreServerModule/EnterpriseReadReplicaEditionModule